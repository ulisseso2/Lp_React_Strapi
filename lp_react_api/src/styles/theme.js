export const theme = {
  colors: {
    primary: '#457B9D',
    light: '#A8DADC',
    dark: '#1D3557',
    hot: '#E63946',
    white: '#f7fff7',
    black: '#343434',
  },
  fonts: {
    family: {
      main: "'Open Sans', sans-serif",
      title: "'Montserrat', sans-serif",
      body: "'Open Sans', sans-serif",
    },
  },
};
//tema padrão da aplicação usado em vários componentes
