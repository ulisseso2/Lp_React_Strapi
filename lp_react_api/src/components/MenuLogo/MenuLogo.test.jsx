import { screen } from '@testing-library/react';
import { MenuLogo } from '.';
import { renderTheme } from '../../styles/render.theme';

describe('<MenuLogo>', () => {
  it('should render text logo', () => {
    renderTheme(<MenuLogo link="#target" text="Olá mundo" />);
    const heading = screen.getByRole('heading', { nome: 'Olá mundo' });
    expect(heading.firstChild).toHaveAttribute('href', '#target');
  });
  it('should render image logo', () => {
    renderTheme(
      <MenuLogo link="#target" text="Olá mundo" srcImg="image.jpg" />,
    );
    const heading = screen.getByRole('heading', { nome: 'Olá mundo' });
    expect(screen.getByRole('img', { name: 'Olá mundo' })).toHaveAttribute(
      'src',
      'image.jpg',
    );
  });
  it('shold match snapshot', () => {
    const { container } = renderTheme(
      <MenuLogo link="#tardet" text="Olá Mundo" srcImg="image.jpg" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
