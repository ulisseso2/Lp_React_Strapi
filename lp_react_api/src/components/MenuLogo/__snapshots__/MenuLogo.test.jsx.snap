// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLogo> shold match snapshot 1`] = `
<h1
  class="sc-bczRLJ kukvte"
>
  <a
    class="sc-gsnTZi gAnVRD"
    href="#tardet"
  >
    <img
      alt="OlÃ¡ Mundo"
      src="image.jpg"
    />
  </a>
</h1>
`;
